app/main.py

```diff
@@ -1,2 +1,72 @@
+from __future__ import annotations
+from typing import Any
+
+
 class Dictionary:
-    pass
+    def __init__(self) -> None:
+        self.length = 0
+        self.hash_table: list = [None] * 8
+
+    def resize_hash_table(self) -> bool:
+        if self.length >= int(len(self.hash_table) * 2 / 3):
+            self.hash_table += [None] * len(self.hash_table)
+
+            return True
+        return False
+
+    def reindex_element_after_resize_table(self) -> None:
+        elements_before_resize = [
+            element
+            for element in self.hash_table
+            if isinstance(element, tuple)
+        ]
+        self.hash_table = [None] * len(self.hash_table)
+
+        for key, value in elements_before_resize:
+            self.insert_element_in_hash_table(key=key, value=value)
+
+    def insert_element_in_hash_table(self, key: Any, value: Any) -> None:
```

Comment:
For the key parameter, a more appropriate annotation would be `Hashable`

Reply:
👍